### PR TITLE
 [ELY-816][ELY-867] masked passwords - initialization vector + XML config

### PR DESCRIPTION
**Pull request by hkalina**
* Support of initialization vector into MaskedPassword: https://issues.jboss.org/browse/ELY-867
* Support of MaskedPassword into XML: https://issues.jboss.org/browse/ELY-816
* Compatibility check reduced to binary compatibility: https://issues.jboss.org/browse/ELY-1572 (because of adding default method into interface)
*XML changes - new XSD - 1.3: https://issues.jboss.org/browse/ELY-1662

### CODE REVIEW

File: src/main/java/org/wildfly/security/password/spec/MaskedPasswordAlgorithmSpec.java [hunk 1]
```diff
@@ -75,16 +94,26 @@ public int getIterationCount() {
...
+
     public boolean equals(Object other) {
         if (! (other instanceof MaskedPasswordAlgorithmSpec)) return false;
         MaskedPasswordAlgorithmSpec otherSpec = (MaskedPasswordAlgorithmSpec) other;
         return otherSpec == this
             || Arrays.equals(initialKeyMaterial, otherSpec.initialKeyMaterial)
+            && iterationCount == otherSpec.iterationCount
             && Arrays.equals(salt, otherSpec.salt)
-            && iterationCount == otherSpec.iterationCount;
+            && initializationVector == otherSpec.initializationVector;
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/password/spec/MaskedPasswordAlgorithmSpec.java]: This should use Arrays.equals()

Comment (hkalina) [reply src/main/java/org/wildfly/security/password/spec/MaskedPasswordAlgorithmSpec.java]: sry, fixed

**Comment by hkalina** [comment]
wait a sec, I need to remove one temporary test yet...

**Comment by hkalina** [comment]
ok now (removed MaskedPasswordTest.test() which I was using to generate reference input for xml)

**Comment by hkalina** [comment]
Fixed MaskedPasswordAlgorithmParameters encoding/decoding to be backward compatible.

**Comment by mchoma** [comment]
Can we discuss ELY-816 a bit? Is it some alternative to credential store? Anyway as it is exposing new configuration options into wildfly-config.xml I think this is in area of RFE. Needs analysis document/tests/documentation. 

Also can you elaborate why some services were removed from WildFlyElytronProvider.java. Is it safe? Are we sure user don't use them?

**Comment by hkalina** [comment]
It is a bit similar to credential-store - while the credential-store allow to move the password into external file, the masked password allow to obfuscate it in place. It is replacement of vault password obfuscation from picketbox.
Also the credential-store reference needs a password for the external file - and that password needs to be directly in the XML in the end - masked password can be used to obfuscate it. (EDIT: this is missing yet, I should add masked password as alternative to clear-text in credential-store-reference-type yet)

services removed from WildFlyElytronProvider: This algorithms are not usable for given purpose - trying to use them for the encryption lead to an exception -> safe to remove. (keeping constants for binary compatibility)

**Comment by darranl** [comment]
This one is going to need a rebase but I think we also need to handle as an RFE so lets pick this up next week.

I would also like to double check if we have anything else going to add a new client side schema, I don't mind bumping as we need to but would still be nice to group as much in at once.

**Comment by hkalina** [comment]
rebased

**Comment by grgrzybek** [comment]
@hkalina any chance this PR gets merged? Any problems with this PR (except conflicts)?

**Comment by hkalina** [comment]
@grgrzybek no idea, ask @darranl
PR is waiting for accepting by reviewers

**Comment by darranl** [comment]
As an enhancement this is currently being considered along side the other priorities we have for WildFly 17, we will be reviewing again shortly.

**Comment by grgrzybek** [comment]
Thanks for info.

**Comment by darranl** [comment]
Just added the hold label as soon we will port this to the latest Elytron version.

**Comment by ashley-abdelsayed98** [comment]
This PR is superseded by #1297

**Pull request closed without merging**